cleverhans/experimental/certification/utils.py

```diff
@@ -216,8 +218,11 @@ def tf_lanczos_smallest_eigval(vector_prod_fn,
   q_vectors = tf.TensorArray(
       dtype, size=1, dynamic_size=True, element_shape=(matrix_dim, 1))
 
-  # Create random vector with Euclidean norm 1
-  b = tf.random_normal(shape=(matrix_dim, 1), dtype=dtype)
+  # If start vector is all zeros, make it a random normal vector and run for 1000 iter
+  if tf.reduce_sum(b) == 0:
+    b = tf.random_normal(shape=(matrix_dim, 1), dtype=dtype)
+    max_iter = 1000
```

Comment:
This value should be an argument of the function.

If you decide to go route 1 (from the comment above) then you don't need this at all.

If you decide to go route 2 (from the comment above) then you can do one of the following
* provide two arguments with number of iterations: `num_iter_full`, `num_iter_warm_start` and choose proper one inside the function
* have only one argument `max_iter`, but when calling this function make it equal to something like `tf.cond(use_warm_starts, lambda: tf.constant(100), lambda: tf.constant(1000))`

If you decide to go route 3, then probably you should have two arguments `num_iter_full`, `num_iter_warm_start`